flake.nix

```diff
@@ -0,0 +1,121 @@
+{
+  description = "PHP development environments";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    nix-shell.url = "github:loophp/nix-shell";
+    systems.url = "github:nix-systems/default";
+    # This should stay until Box 4.5.1 is not on `nixos-unstable` branch
+    nixpkgs-master.url = "github:NixOS/nixpkgs/master";
+  };
+
+  outputs = inputs@{ self, flake-parts, systems, ... }: flake-parts.lib.mkFlake { inherit inputs; } {
+    systems = import systems;
+
+    perSystem = { config, self', inputs', pkgs, system, lib, ... }:
+      let
+        php = pkgs.api.buildPhpFromComposer {
+          src = ./.;
+          php = pkgs.box451.php81; # Change to php56, php70, ..., php81, php82, php83 etc.
+        };
+      in
+      {
+        _module.args.pkgs = import self.inputs.nixpkgs {
+          inherit system;
+          overlays = [
+            inputs.nix-shell.overlays.default
+            (final: prev: {
+              box451 = import inputs.nixpkgs-master {
+                inherit system;
+              };
+            })
+          ];
+          config.allowUnfree = true;
+        };
+
+        apps = {
+          default = {
+            type = "app";
+            program = lib.getExe self'.packages.backwardcompatibilitycheck;
+          };
+
+          build-phar = {
+            type = "app";
+            program = lib.getExe self'.packages.build-phar-script;
+          };
+        };
+
+        checks = {
+          inherit (self'.packages) phar;
+        };
+
+        devShells.default = pkgs.mkShellNoCC {
+          name = "php-devshell";
+          buildInputs = [
+            php
+            php.packages.composer
+            php.packages.box
+          ];
+        };
+
+        packages = {
+          backwardcompatibilitycheck = php.buildComposerProject {
+            pname = "backwardcompatibilitycheck";
+            version = "8.x.x-dev";
+
+            src = ./.;
+
+            # This only changes when `composer.lock` is updated
+            vendorHash = "sha256-LsrGmver7RyiI0/l2j6dZaqhFQf2OFyUOZb8xzFFEIA=";
```

Comment:
Ok, so we're back at having the hash here: how would one approach a patch by Renovate where `composer.lock` is being touched? :thinking:

Reply:
Yeah... I can't basically get rid of it, I tried... and there's a reason to have it there.

Since this PR is quite messy, how about lively discuss this during an informal meeting at your best convenience?

Reply:
Note: could be worked around by assuming `--prefer-source` to download via `git`, where we at least have a SHA1